### PR TITLE
Softmax and Layer_Norm Backward Support

### PR DESCRIPTION
**Pull request by rdspring1**
- [X] Create Softmax Benchmark
- [X] Create Batch_Norm Benchmark
- [X] Create Layer_Norm Benchmark
- [X] Add Backwards pass for Softmax
- [X] Add Backwards pass for Layer_Norm
- [X] Verified with MNIST example

### CODE REVIEW

**Comment by rdspring1** [comment]
I changed the `native_layer_norm` and `native_layer_norm_backward` signatures, so they match the external `aten::layer_norm` definition now.

**Review by jjsjann123 — APPROVED** [review]
ALL TESTS ARE PASSING!!! MERGING THIS!!!
